src/memory-model.md

```diff
@@ -1,5 +1,26 @@
+r[memory]
 # Memory model
 
-Rust does not yet have a defined memory model. Various academics and industry professionals
-are working on various proposals, but for now, this is an under-defined place
-in the language.
+> [!WARNING]
+> The memory model of Rust is incomplete and not fully decided.
+
+r[memory.bytes]
+## Bytes
+
+r[memory.bytes.intro]
+The most basic unit of memory in Rust is a byte. All values are computed from zero or more bytes read from an allocation.
+
+> [!NOTE]
+> While bytes are typically lowered to hardware bytes, Rust uses an "abstract" notion of bytes that can make distinctions which are absent in hardware, such as being uninitialized, or storing part of a pointer.
+
+r[memory.bytes.contents]
+Each byte may have one of the following values:
+
+r[memory.bytes.init]
+* An initialized byte containing a `u8` value and optional [provenance][std::ptr#provenance],
+
+r[memory.bytes.uninit]
+* An uninitialized byte.
```

Comment:
I find this enumeration interrupted by `r[...]` quite hard to read. But maybe that can be fixed with better stylesheets? There was some discussion about moving the `r` headers to the left of the text if there is enough horizontal space, is that going to happen?

Reply:
It already did.  You should be seeing it.

Reply:
Ah, I looked at https://doc.rust-lang.org/reference/behavior-considered-undefined.html. The [nightly](https://doc.rust-lang.org/nightly/reference/behavior-considered-undefined.html) version looks a lot better indeed. :)